src/compiler.rs

```diff
@@ -199,10 +199,30 @@ impl AmberCompiler {
         }
 
         let now = Local::now().format("%Y-%m-%d %H:%M:%S").to_string();
-        let header = include_str!("header.sh")
+        
+        let header_template = 
+            if let Ok(dynamic) = env::var("AMBER_HEADER") {
+                fs::read_to_string(dynamic.to_string()).expect(format!("Couldn't read the dynamic header file from {dynamic}").as_str())
+            } else {
+                include_str!("header.sh").to_string()
+            };
+        
+        let footer_template = 
+            if let Ok(dynamic) = env::var("AMBER_FOOTER") {
+                fs::read_to_string(dynamic.to_string()).expect(format!("Couldn't read the dynamic footer file from {dynamic}").as_str())
+            } else {
+                String::new()
+            };
+        
+        let header = header_template
+            .replace("{{ version }}", env!("CARGO_PKG_VERSION"))
+            .replace("{{ date }}", now.as_str());
+        
+        let footer = footer_template
```

Comment:
This section is pretty long and adds up to the length of the `translate` method. Would you consider extracting it to a new private `gen_header` (or something similar) method?